Add a generator for an Extension pack

egamma opened the pull request:
Addresses https://github.com/Microsoft/vscode-generator-code/issues/63.

aeschli commented:
- when running against vscode without any extension installed, I get the following package.json
    "extensionDependencies": [ 
        ""
    ]
}
- disabled extensions are also listed
- When I don't have 'code' installed I get
Error: Uncaught, unspecified "error" event. (undefined)
    at emit (events.js:157:17)
    at Immediate._onImmediate (/home/aeschli/workspaces/vscode-generator-code/node_modules/yeoman-generator/lib/base.js:446:16)
    at tryOnImmediate (timers.js:534:15)
    at processImmediate [as _immediateCallback] (timers.js:514:5)

I suggest to print an error to the console instead and continue.
- For completeness, there should also be a test case (at least for the 'no' (don't read existing)) case. See https://github.com/Microsoft/vscode-generator-code/blob/master/test/test.js.

egamma commented:
> when running against vscode without any extension installed, I get the following package.json

Fixed

> disabled extensions are also listed

Currently there is no such command line option. Filed https://github.com/Microsoft/vscode/issues/19083

> When I don't have 'code' installed I get

Fixed.

> For completeness, there should also be a test case (at least for the 'no' (don't read existing)) case. 

Added test.

Also added a warning at the end that user reviews the `extensionDependencies` attribute.

egamma commented:
@aeschli pls review again, the generator should now be in good shape.

aeschli commented on generators/app/index.js:
```diff
@@ -304,6 +310,41 @@ module.exports = yeoman.Base.extend({
+                        childProcess.exec('code --list-extensions', function(error, stdout, stderr) {
+                            if (error) {
+                                generator.env.error("Problems starting Code: " + error);
+                                return;
```
'resolve' should be called before the return

aeschli commented on test/test.js:
```diff
@@ -280,4 +280,42 @@ describe('test theme generator', function () {
+                name: 'testExtensionPack',
+                displayName: 'Test Extension Pack',
+                description: 'My Test Extension Pack',
+                publisher: 'Microsoft'
```
Add 'addExtensions' here with true or false

aeschli replied:
Both variants (true and false) are interesting as test case.

aeschli commented:
PR merged! Thanks @egamma!

egamma commented:
Thanks @aeschli for the careful review and guidance.
